tasks/test_results_processor.py

```diff
@@ -0,0 +1,201 @@
+                log.warning(f"Error parsing testruns")
+                return {"successful": False}
+
+            # concat existing and new test information
+            testrun_list += parsed_testruns
+
+            upload_list.append(upload_obj)
+
+        # save relevant stuff to database
+        self.save_report(db_session, testrun_list, upload_obj)
+
+        # kick off notification task stuff
+
+        if self.should_delete_archive(commit_yaml):
+            repository = (
+                db_session.query(Repository)
+                .filter(Repository.repoid == int(repoid))
+                .first()
+            )
+            self.delete_archive(
+                commitid, repository, commit_yaml, uploads_to_delete=upload_list
+            )
+
+        return {
+            "successful": True,
+            "testrun_list": [self.testrun_to_dict(t) for t in testrun_list],
+        }
+
+    def save_report(
+        self, db_session: Session, testrun_list: List[Testrun], upload: Upload
+    ):
+        for testrun in testrun_list:
+            test = (
+                db_session.query(Test)
+                .filter_by(name=testrun.name, testsuite=testrun.testsuite)
+                .first()
+            )
+            if not test:
+                test = Test(
+                    repoid=upload.report.commit.repoid,
+                    name=testrun.name,
+                    testsuite=testrun.testsuite,
+                )
+                db_session.add(test)
+                db_session.flush()
+
+            db_session.add(
+                TestInstance(
+                    test_id=test.id,
+                    duration_seconds=testrun.duration,
+                    outcome=int(testrun.outcome),
+                    upload_id=upload.id,
+                )
+            )
+            db_session.flush()
+
+    def process_individual_arg(self, upload: Upload, repository) -> List[Testrun]:
+        archive_service = ArchiveService(repository)
+
+        payload_bytes = archive_service.read_file(upload.storage_path)
+        data = json.loads(payload_bytes)
+
+        testrun_list = []
+
+        for file in data["test_results_files"]:
+            file = file["data"]
+            print(file)
+            file_bytes = BytesIO(zlib.decompress(base64.b64decode(file)))
+            print(file_bytes)
+
+            first_line = file_bytes.readline()
+            file_bytes.seek(0)
+
+            file_content = file_bytes.read()
+
+            # TODO: improve report matching capabilities
+            # use file extensions?
+            # maybe do the matching in the testing result parser lib?
+            try:
+                if first_line.startswith(b"<?xml"):
+                    testrun_list = parse_junit_xml(file_content)
+                elif b"pytest" in first_line:
+                    testrun_list = parse_pytest_reportlog(file_content)
+                else:
+                    testrun_list = parse_vitest_json(file_content)
```

Comment:
Should we maybe match something for vitest and then "else" raise a not supported error? 

This might get weird otherwise.